src/java.base/share/classes/java/lang/invoke/IndirectVarHandle.java

```diff
@@ -61,33 +60,31 @@ private IndirectVarHandle(VarHandle target, Class<?> value, Class<?>[] coordinat
         super(form, exact);
         this.handleFactory = handleFactory;
         this.target = target;
-        this.directTarget = target.asDirect();
         this.value = value;
         this.coordinates = coordinates;
     }
 
     @Override
-    public Class<?> varType() {
-        return value;
+    MethodType accessModeTypeUncached(AccessType at) {
+        return at.accessModeType(null, value, coordinates);
     }
 
     @Override
-    public List<Class<?>> coordinateTypes() {
-        return List.of(coordinates);
-    }
+    public boolean isAccessModeSupported(AccessMode accessMode) {
+        var directTarget = this.directTarget;
```

Comment:
Why is this not defined in the superclass, and then use `asDirect` (as done in other cases) ?

Reply:
I aim to avoid eager evaluation of directTarget (which is lazy in LazyStaticVarHandle, computing it requires initializing the holder class). This code path checks a stable field, so it should allow constant-folding when the direct target is available.

Reply:
I see - so, since this is just a "query" you would like to avoid initialization just for this.